bin/modules/calculate_properties.py

```diff
@@ -345,3 +349,590 @@ def getStructuresFromOptimization(outputfolder, FrameworkName) -> list:
+
+    vol = np.dot(v1, np.cross(v2, v3))
+
+    b1 = 2 * np.pi * np.cross(v2, v3) / vol
+    b2 = 2 * np.pi * np.cross(v3, v1) / vol
+    b3 = 2 * np.pi * np.cross(v1, v2) / vol
+
+    return b1, b2, b3
+
+
+def get_kgrid(cell, dist=0.3) -> tuple[float, float, float]:
+    '''Get the k-points grid in the reciprocal space with a given distance for a
+    cell given in cell parameters of cell vectors.
+    ----------
+    cell : array
+        (3,1) array for cell vectors or (6,1) array for cell parameters
+    distance : float
+        distance between the points in the reciprocal space
+    Returns
+    -------
+    kx : int
+        Number of points in the x direction on reciprocal space
+    ky : int
+        Number of points in the y direction on reciprocal space
+    kz : int
+        Number of points in the z direction on reciprocal space
+    '''
+
+    b1, b2, b3 = get_reciprocal_vectors(cell)
+
+    b = np.array([np.linalg.norm(b1),
+                  np.linalg.norm(b2),
+                  np.linalg.norm(b3)])
+
+    kx = np.ceil(b[0]/dist).astype(int)
+    ky = np.ceil(b[1]/dist).astype(int)
+    kz = np.ceil(b[2]/dist).astype(int)
+
+    return kx, ky, kz
+
+
+def create_input_file(FrameworkName: str,
+                      output_folder: str,
+                      **kwargs):
+    """
+    Create the input file for CP2K
+
+    Parameters
+    ----------
+    FrameworkName : str
+        Name of the framework
+    output_folder : str
+        Path to the output folder
+    """
+
+    CalcDict = {
+        'FrameworkName': FrameworkName.split('.')[0],
+        'Charge': 0,
+        'Multiplicity': 1,
+        'CalcType': 'energy_force',  # Can be 'energy_force', 'cell_opt', 'geo_opt', 'md', or 'normal_modes'
+        'UseOT': False,
+        'UseSmearing': False,
+        'SmearingMethod': 'fermi_dirac',  # Can be 'fermi_dirac' or 'energy_window'
+        'ElectronicTemperature': 300,
+        'WindowSize': 0.1,
+        'AddedMOs': 0,
+        'MixingMethod': 'broyden_mixing',  # Can be 'direct_p_mixing', 'broyden_mixing_new', or 'kerker_mixing'
+        'MixingAlpha': 0.2,
+        'MaxSCFycles': 30,
+        'MaxOuterSCFycles': 10,
+        'EPSDefault': 1e-8,
+        'PWCutoff': 1200,
+        'NGrid': 5,
+        'RelativeCutOff': 60,
+        'Functional': 'PBE',  # Can be 'PBE', 'XTB', or 'PBE0'
+        'Parametrization': 'ORIG',  # Can be 'ORIG', 'PBESOL', or 'REVPBE'
+        'DispersionCorrection': 'DFTD3',  # Can be None, 'DFTD2', 'DFTD3', or 'DFTD3(BJ)'
+        'CheckAtomicCharges': True,
+        'BasisSet': 'DZVP',  # Can be 'DZVP', 'TZVP', or 'TZV2P'
+        'SCFGuess': 'atomic',  # Can be 'atomic', 'restart', 'core', 'random', 'sparse', or 'mopac'
+        'SCFConvergence': 1e-8,
+        'CP2KDataDir': os.environ.get("CP2K_DATA_DIR"),
+        'KeepSymmetry': False,
+        'KeepSpaceGroup': False,
+        'KeepAngles': False,
+        'MaxIterations': 100,
+        'Restart': False,
+        'MaxIterations': 100,
```

Comment:
This argument appears twice. If you sort them alphabetically, it will be easier to spot these duplicates.

Reply:
Done